sdk/storage/azure-storage-blob/azure/storage/blob/_shared/xml_deserialization.py

```diff
@@ -0,0 +1,587 @@
+
+    subtype_xml_map = getattr(subtype, "_xml_map", {})
+    xml_name = subtype_xml_map.get('name', xml_name)
+    return data.find(xml_name)
+
+
+class Deserializer(object):
+    """Response object model deserializer.
+
+    :param dict classes: Class type dictionary for deserializing complex types.
+    """
+    def __init__(self, classes=None):
+        self.deserialize_type = {
+            'str': deserialize_unicode,
+            'int': deserialize_int,
+            'bool': deserialize_bool,
+            'float': deserialize_float,
+            'iso-8601': deserialize_iso,
+            'rfc-1123': deserialize_rfc,
+            'unix-time': deserialize_unix,
+            'duration': deserialize_duration,
+            'date': deserialize_date,
+            'time': deserialize_time,
+            'decimal': deserialize_decimal,
+            'long': deserialize_long,
+            'bytearray': deserialize_bytearray,
+            'base64': deserialize_base64,
+            'object': deserialize_object,
+            '[]': self.deserialize_iter,
+            '{}': self.deserialize_dict
+            }
+
+        self.dependencies = dict(classes) if classes else {}
+
+    def __call__(self, target_obj, response_data, **kwargs):
+        """Call the deserializer to process a REST response.
+
+        :param str target_obj: Target data type to deserialize to.
+        :param requests.Response response_data: REST response object.
+        :param str content_type: Swagger "produces" if available.
+        :raises: DeserializationError if deserialization fails.
+        :return: Deserialized object.
+        """
+        try:
+            # First, unpack the response if we have one.
+            response_data = unpack_xml_content(response_data.http_response, **kwargs)
+        except AttributeError:
+            pass
+        if response_data is None:
+            # No data. Moving on.
+            return None
+        #return self._deserialize(target_obj, response_data)
+        return self.deserialize_data(response_data, target_obj)
+
+    def failsafe_deserialize(self, target_obj, data, content_type=None):
+        """Ignores any errors encountered in deserialization,
+        and falls back to not deserializing the object. Recommended
+        for use in error deserialization, as we want to return the
+        HttpResponseError to users, and not have them deal with
+        a deserialization error.
+
+        :param str target_obj: The target object type to deserialize to.
+        :param str/dict data: The response data to deseralize.
+        :param str content_type: Swagger "produces" if available.
+        """
+        try:
+            return self(target_obj, data, content_type=content_type)
+        except:  # pylint: disable=bare-except
+            _LOGGER.warning(
+                "Ran into a deserialization error. Ignoring since this is failsafe deserialization",
+				exc_info=True
+            )
+            return None
```

Comment:
It seems this not decoding the response body? ContentDecodingPolicy was turning body which describes the error into string, so probably we want to do the same thing?

Reply:
also some response body on error is in json format, not sure if it will be a problem